packages/api/lambdas/sf-starter.js

```diff
@@ -27,33 +42,140 @@ function dispatch(message) {
 }
 
 /**
- * This is an SQS Queue consumer.
+ * Increment the priority semaphore.
+ *
+ * @param {string} key - Key for the priority semaphore
+ * @param {number} maximum - Maximum number of executions allowed for this semaphore
+ * @returns {Promise}
+ * @throws {Error}
+ */
+async function incrementPrioritySemaphore(key, maximum) {
+  const semaphore = new Semaphore(
+    dynamodbDocClient(),
+    process.env.SemaphoresTable
+  );
+
+  try {
+    await semaphore.up(key, maximum);
+  } catch (err) {
+    if (err instanceof ResourcesLockedError) {
+      log.info(`The maximum number of executions for ${key} are already running. Could not start a new execution.`);
+    }
+    throw err;
+  }
+}
+
+/**
+ * Attempt to increment the priority semaphore and start a new execution.
+ *
+ * If `incrementPrioritySemaphore()` is unable to increment the priority semaphore,
+ * it throws an error and `dispatch()` is not called.
+ *
+ * @param {Object} queueMessage - SQS message
+ * @returns {Promise} - Promise returned by `dispatch()`
+ * @throws {Error}
+ */
+async function incrementAndDispatch(queueMessage) {
+  const message = get(queueMessage, 'Body', {});
+
+  if (!has(message, 'cumulus_meta.priorityKey')
+      || isNil(message.cumulus_meta.priorityKey)) {
+    throw new Error('Could not find priority key for message');
+  }
+
+  const priorityKey = get(message, 'cumulus_meta.priorityKey');
```

Comment:
If this is moved above the `if` statement, the `if` statement can be simplified to `if (isNil(priorityKey))`